src/dune_engine/vcs.ml

```diff
@@ -79,9 +79,22 @@ let run t args =
   in
   String.trim s
 
-let run_zero_separated t args =
-  Process.run_capture_zero_separated Strict (prog t) args ~dir:t.root
-    ~env:Env.initial
+let git_accept () =
+  Process.Accept (Predicate_lang.union [ Element 0; Element 128 ])
+
+let run_git t args =
+  let res =
+    Process.run_capture (git_accept ()) (prog t) args ~dir:t.root
+      ~env:Env.initial
+      ~stderr_to:(Process.Io.file Config.dev_null Out)
+  in
+  let open Fiber.O in
+  let+ res = res in
+  match res with
+  | Ok s -> Some (String.trim s)
+  | Error 128 -> None
+  | Error n ->
+    User_error.raise [ Pp.textf "Unexpected response from `git` (code %d)" n ]
```

Comment:
I believe this is unreachable, i.e. `Process` will already raise in this case. So `assert false` might be enough